op.c

```diff
@@ -898,6 +898,11 @@ Perl_op_free(pTHX_ OP *o)
         /* free child ops before ourself, (then free ourself "on the
          * way back up") */
 
+        /* Detect the caller being silly
+         *   https://github.com/Perl/perl5/issues/20764
+         */
```

Comment:
I think the comment could be improved.  Perhaps:

> Ensure the caller maintains the relationship between OPf_KIDS and op_first when restructuring the tree.

Reply:
Ahyes; message rewritten and force-pushed.